### PR TITLE
Refactor and add PyTorch models

### PR DESCRIPTION
**Pull request by zphang**
Will rebase once changes are finalized. Should not merge until after rebase.

Changes:

* Added PyTorch as well as CPU options
* Added tests
* Code style changes
* Uploaded PyTorch models

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,70 @@
...
+This is an implementation of the model used for breast density classification as described in our paper ["Breast density classification with deep convolutional neural networks"](https://arxiv.org/pdf/1711.03674.pdf). The implementation allows users to get the breast density prediction by applying one of our pretrained models: a histogram-based model or a multi-view CNN. Both models act on screening mammography exams with four standard views. As a part of this repository, we provide a sample exam (in `images` directory). The models are implemented in both TensorFlow and PyTorch.
 
 ## Prerequisites
 
-* Python (3.6), TensorFlow (1.5.0), NumPy (1.14.3), SciPy (1.0.0).
-* NVIDIA GPU (we used Tesla M40).
```
Comment (wooginawunan) [inline README.md]: Please keep the version of NVIDIA GPU in the readme.md

File: density_model_tf.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import argparse
+import tensorflow as tf
+
+import models_tf as models
+import utils
+
+
+def optimistic_restore(session, save_file):
```
Comment (wooginawunan) [inline density_model_tf.py]: Is it possible to move this function (optimistic_restore) into utils?

File: README.md [hunk 3]
```diff
@@ -1,42 +1,70 @@
...
 
-With `model = 'histogram'`:
+With `histogram`:
 ```
 Density prediction:
-        Almost entirely fatty (0):                      0.08194443
-        Scattered areas of fibroglandular density (1):  0.7830397
-        Heterogeneously dense (2):                      0.13350312
-        Extremely dense (3):                            0.0015126525
+        Almost entirely fatty (0):                      0.0819444
```
Comment (wooginawunan) [inline README.md]: Problem with indentation.

File: density_model_tf.py (hunk 2 above)
Comment (zphang) [reply density_model_tf.py]: optimistic_restore is library specific, so I think it should be kept within the TensorFlow files. We can move it to models though.

File: README.md (hunk 1 above)
Comment (zphang) [reply README.md]: GPU usage is now optional, but I can add it back if you think it’s necessary.

**Pull request closed without merging**